test/vtt-segment-loader.test.js

```diff
@@ -676,17 +676,38 @@ QUnit.module('VTTSegmentLoader', function(hooks) {
           'requesting initial segment guess'
         );
 
+        // set the pending segment to mediaIndex 1
+        // so that the next request will attempt to grab this empty segment.
+        loader.pendingSegment_.mediaIndex = 1;
+
         this.requests[0].responseType = 'arraybuffer';
         this.requests.shift().respond(200, null, new Uint8Array(10).buffer);
 
         this.clock.tick(1);
 
         assert.ok(playlist.segments[2].empty, 'marked empty segment as empty');
+
         assert.equal(
           loader.pendingSegment_.mediaIndex,
           3,
           'walked forward skipping requesting empty segment'
         );
+
+        // set the pending segment to mediaIndex 1 again
+        // so that the next request will attempt to grab the next two empty segments.
+        loader.pendingSegment_.mediaIndex = 1;
+
+        this.requests[0].responseType = 'arraybuffer';
+        this.requests.shift().respond(200, null, new Uint8Array(10).buffer);
+
+        this.clock.tick(1);
+
+        assert.ok(playlist.segments[3].empty, 'marked empty segment as empty');
+
+        assert.ok(
+          !loader.pendingSegment_,
+          'no pending segment, and no error even though last two segments are empty'
```

Comment:
If there was an error I think it would be on the `loader` itself, which may be worth checking for the second part of this message.